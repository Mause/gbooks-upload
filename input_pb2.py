# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: input.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import builder as _builder
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import timestamp_pb2 as google_dot_protobuf_dot_timestamp__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x0binput.proto\x1a\x1fgoogle/protobuf/timestamp.proto\"S\n\x03Tag\x12\x0c\n\x04name\x18\x01 \x02(\t\x12\x0e\n\x06tag_id\x18\x02 \x02(\t\x12.\n\ncreated_at\x18\x03 \x01(\x0b\x32\x1a.google.protobuf.Timestamp\"X\n\x06Tagged\x12\x0f\n\x07\x62ook_id\x18\x01 \x02(\t\x12\x0e\n\x06tag_id\x18\x02 \x02(\t\x12-\n\ttagged_at\x18\x03 \x02(\x0b\x32\x1a.google.protobuf.Timestamp\"A\n\x0cTagsResponse\x12\x12\n\x04tags\x18\x01 \x03(\x0b\x32\x04.Tag\x12\x1d\n\x0ctagged_items\x18\x02 \x03(\x0b\x32\x07.Tagged\"+\n\nTagRequest\x12\x1d\n\x0ctagged_items\x18\x01 \x02(\x0b\x32\x07.Tagged')

_globals = globals()
_builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, _globals)
_builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'input_pb2', _globals)
if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  _globals['_TAG']._serialized_start=48
  _globals['_TAG']._serialized_end=131
  _globals['_TAGGED']._serialized_start=133
  _globals['_TAGGED']._serialized_end=221
  _globals['_TAGSRESPONSE']._serialized_start=223
  _globals['_TAGSRESPONSE']._serialized_end=288
  _globals['_TAGREQUEST']._serialized_start=290
  _globals['_TAGREQUEST']._serialized_end=333
# @@protoc_insertion_point(module_scope)
