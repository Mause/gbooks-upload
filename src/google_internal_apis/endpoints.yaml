playatoms-pa.clients6.google.com:
  "google.internal.play.atoms.api.v1.WhApiService":
    methods:
      - method: GetAppSplits
      - method: UpdateUserPrefs
      - method: GetDomainFilter
      - method: GetIntentFilter

calendar-pa.clients6.google.com:
  google.internal.calendar.v1.AppointmentBookingService:
    methods:
      - method: ListAvailableSlots
        data: [null,null,"cal_id",null,[['from_timestamp'],['end_timestamp']]]

instantmessaging-pa.googleapis.com:
  google.internal.communications.instantmessaging.v1.Pairing:
    methods:
      - method: RegisterPhoneRelay
      - method: RefreshPhoneRelay
      - method: GetWebEncryptionKey
      - method: RevokeRelayPairing

  google.internal.communications.instantmessaging.v1.Messaging:
    methods:
      - method: ReceiveMessages
      - method: SendMessage
      - method: AckMessages

  google.internal.communications.instantmessaging.v1.Registration:
    methods:
      - method: SignInGaia
      - method: RegisterRefresh

meet.google.com:
  google.rtc.meetings.v1.MeetingDeviceService:
    methods:
      - method: CreateMeetingDevice

  google.rtc.meetings.v1.MeetingSpaceService:
    methods:
      - method: SyncMeetingSpaceCollections

people-pa.clients6.google.com:
  google.internal.people.v2.InternalPeopleService:
    methods:
    - method: GetPeople
      body: '[["me"],[[["person.name","person.email","person.photo"]],null,[5,1,7]]]'
    - method: GetPersonPhotoEncoded
      body: null
    - method: UpdatePersonPhoto
      body: null

  google.internal.people.v2.minimal.InternalPeopleMinimalService:
    methods:
      - method: ListRankedTargets
      - method: ListPeopleByKnownId
        data: [
            [email],
            2,
            None,
            None,
            [None, None, None, ["DRIVE_SHARE", None, 2]],
            [
                [
                    [
                        "person.name",
                        "person.photo",
                        "person.email",
                        "person.metadata",
                        "person.name.metadata.verified",
                        "person.email.metadata.verified",
                    ]
                ],
                None,
                [2, 8, 1, 7, 10, 11],
                None,
                None,
                [],
            ],
            [],
            [None, None, [], None, None, [1], [[1]], None, [3]],
            [[23, 36]],
            [],
        ]

playbooks-pa.clients6.google.com:
  google.internal.play.books.audiobook.v1.AudiobookService:
    methods:
    - method: GetAudioEvents
      body: '["AQAAAEAszz6PAM","8af46ef0d127346baff59518",[8],4976830,6776830]'
    - method: GetAudiobookResource
      body: null
    - method: GetAudiobookSupplement
      body: null
  google.internal.play.books.cloudloading.v1.CloudLoadingOnePlatformService:
    methods:
    - method: Delete
      body: null
    - method: Insert
      body: null
  google.internal.play.books.enterprise.v1.EnterpriseService:
    methods:
    - method: AddClassroomBooksAttachment
      body: null
    - method: AddLicenseOwners
      body: null
    - method: AddLicenseRedeemers
      body: null
    - method: ClaimLicense
      body: null
    - method: GetLicenseHistory
      body: null
    - method: GetLicenses
      body: '[null,null,null,1]'
    - method: GrantFreeBookLicenses
      body: null
    - method: ListFreeBooksForLicense
      body: null
    - method: RemoveLicenseOwners
      body: null
    - method: RemoveLicenseRedeemers
      body: null
  google.internal.play.books.family.v1.FamilyOnePlatformService:
    methods:
    - method: Share
      body: null
    - method: Unshare
      body: null
  google.internal.play.books.layers.v1.VolumeAnnotationService:
    methods:
    - method: GetDictionaryDefinition
      body: '[null," unwrapping",[null,null,"en"]]'
  google.internal.play.books.library.v1.LibraryService:
    methods:
    - method: AddAnnotation
      body: '[["AQAAAEAszz6PAM","AE67D741-9B6D-4C7D-B170-312AF12BE063",[],null,null,[["40102"]],"0.10.0"]]'
    - method: AddLibraryDocument
      body: null
    - method: AddTags
      body: '[[["5QuwNwAAAEAJ","a0c746137-5723-4740-89a8-4ef7311aad16"]]]'
      error: [
        3,
        "Invalid value at 'add_tags[0].update_timestamp_ms' (TYPE_INT64), \"1737106632509.533\"",
        [
            [
                "type.googleapis.com/google.rpc.BadRequest",
                [
                    [
                        [
                            "add_tags[0].update_timestamp_ms",
                            "Invalid value at 'add_tags[0].update_timestamp_ms' (TYPE_INT64), \"1737106632509.533\""
                        ]
                    ]
                ]
            ]
        ]
      ]
    - method: CreateCustomTag
      body: null
    - method: DeleteAnnotation
      body: null
    - method: DeleteCustomTag
      body: null
    - method: DeleteLibraryDocument
      body: null
    - method: ExpungeLibraryDocuments
      body: null
    - method: GetLibraryDocument
      body: '[[], "hdYTCAAAQBAJ"]'
    - method: GetVolumeOverviews
      body: null
    - method: HideLibraryDocuments
      body: null
    - method: ListAnnotations
      body: null
    - method: ListTags
      body: null
    - method: LookupCopyLimit
      body: null
    - method: RemoveTags
      body: '[[["hdYTCAAAQBAJ","Completed",1737103382078]]]'
    - method: SyncDocumentPosition
      body: '["84XKNwAAAEAJ",["384DC38E-1A56-4F48-8AE1-1C3F446F41E5",null,"1737093647557",[null,null,[2]],null,[["GBS.PA68.w.0.0.0.2"],null,0.9066666666666666]]]'
    - method: SyncExtendedLibrary
      body: null
    - method: SyncUserLibrary
      body: null
    - method: UnhideLibraryDocuments
      body: null
    - method: UpdateCopyLimit
      body: null
    - method: UpdateCustomTag
      body: '["a0c746137-5723-4740-89a8-4ef7311aad16","test 1"]'
  google.internal.play.books.series.v1.SeriesOnePlatformService:
    methods:
    - method: Fetch
      body: '[[["u4P3GgAAABBNmM"],["PL4pGwAAABAUpM"],["bW8uGwAAABBCdM"],["i1YuGwAAABCkTM"],["crbMGgAAABC_rM"]]]'
    - method: FetchMembers
      body: '[["u4P3GgAAABBNmM"],[100]]'
  google.internal.play.books.settings.v1.SettingsOnePlatformService:
    methods:
    - method: GetUserSettings
      body: null
    - method: UpdateUserSettings
      body: null
  google.internal.play.books.userannotations.v1.UserAnnotationService:
    methods:
    - method: CreateAnnotations
      body: null
    - method: CreateCollection
      body: null
    - method: DeleteAnnotations
      body: null
    - method: DeleteCollection
      body: null
    - method: ListAnnotations
      body: '[[["84XKNwAAAEAJ",null,"full-1.0.0"]],[1,2],null,1,1]'
    - method: ListSharedAnnotations
      body: '[["84XKNwAAAEAJ",null,"full-1.0.0"],1,null,null,1]'
    - method: TriggerExport
      body: null
    - method: UpdateAnnotations
      body: null
    - method: UpdateCollection
      body: null
    - method: UpdateSubscriptions
      body: null
playgateway-pa.clients6.google.com:
  play.gateway.adapter.books.v1.PlayGatewayBooksService:
    methods:
    - method: GetInAppStream
      body: '[[4,null,["84XKNwAAAEAJ",1]],null,null,null,null,[null,null,null,null,null,null,[3]],null,null,[null,[2],[1]]]'
    - method: UpdateItemWishlistState
      body: '[null,[\"ID07FerN_h0C\",9],1]'

alkalimakersuite-pa.clients6.google.com:
  google.internal.alkali.applications.makersuite.v1.MakerSuiteService:
    methods:
      - method: GenerateContent
      - method: CreateTunedModel

alkalimetricsink-pa.clients6.google.com:
  google.internal.alkali.applications.metricsink.v1.MetricService:
    methods:
      - method: RecordMetrics

content-autofill.googleapis.com: {}
content-developerprofiles-pa.googleapis.com: {}
alkalicore-pa.clients6.google.com: {}
generativelanguage.googleapis.com: {}

addons-pa.clients6.google.com:
  google.internal.apps.addons.v1.AddOnService:
    methods:
      - method: ListInstallations

waa-pa.clients6.google.com:
  google.internal.waa.v1.Waa:
    methods:
    - method: Create
      body: null
    - method: Ping
      body: null
