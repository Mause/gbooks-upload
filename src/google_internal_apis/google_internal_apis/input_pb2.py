# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: google_internal_apis/input.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import builder as _builder
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import timestamp_pb2 as google_dot_protobuf_dot_timestamp__pb2
from google.protobuf import wrappers_pb2 as google_dot_protobuf_dot_wrappers__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n google_internal_apis/input.proto\x1a\x1fgoogle/protobuf/timestamp.proto\x1a\x1egoogle/protobuf/wrappers.proto\"S\n\x03Tag\x12\x0c\n\x04name\x18\x01 \x02(\t\x12\x0e\n\x06tag_id\x18\x02 \x02(\t\x12.\n\ncreated_at\x18\x03 \x01(\x0b\x32\x1a.google.protobuf.Timestamp\"X\n\x06Tagged\x12\x0f\n\x07\x62ook_id\x18\x01 \x02(\t\x12\x0e\n\x06tag_id\x18\x02 \x02(\t\x12-\n\ttagged_at\x18\x03 \x02(\x0b\x32\x1a.google.protobuf.Timestamp\"A\n\x0cTagsResponse\x12\x12\n\x04tags\x18\x01 \x03(\x0b\x32\x04.Tag\x12\x1d\n\x0ctagged_items\x18\x02 \x03(\x0b\x32\x07.Tagged\"+\n\nTagRequest\x12\x1d\n\x0ctagged_items\x18\x01 \x03(\x0b\x32\x07.Tagged\"F\n\x17LibraryDocumentResponse\x12+\n\x11library_documents\x18\x01 \x02(\x0b\x32\x10.LibraryDocument\"\xc0\t\n\x0fLibraryDocument\x12\x0f\n\x07\x62ook_id\x18\x01 \x02(\t\x12#\n\x04\x62ook\x18\x02 \x02(\x0b\x32\x15.LibraryDocument.Book\x1a\xf6\x08\n\x04\x42ook\x12\r\n\x05title\x18\x01 \x02(\t\x12\x0f\n\x07\x61uthors\x18\x02 \x03(\t\x12\x11\n\tpublisher\x18\x03 \x02(\t\x12\x16\n\x0epublished_date\x18\x04 \x02(\t\x12\x13\n\x0b\x64\x65scription\x18\x05 \x02(\t\x12\x12\n\npage_count\x18\x06 \x02(\x05\x12\x0f\n\x07version\x18\x07 \x02(\t\x12\x36\n\x0c\x63over_images\x18\x08 \x03(\x0b\x32 .LibraryDocument.Book.CoverImage\x12\x10\n\x08language\x18\t \x02(\t\x12\x0b\n\x03url\x18\n \x02(\t\x12\x10\n\x08unknown1\x18\x0b \x02(\x05\x12\x11\n\tunknown2s\x18\x0c \x03(\x05\x12\x10\n\x08unknown3\x18\r \x02(\x05\x12\x10\n\x08unknown4\x18\x0e \x02(\x05\x12,\n\x06series\x18\x0f \x02(\x0b\x32\x1c.LibraryDocument.Book.Series\x12\x11\n\tunknown5s\x18\x10 \x03(\x05\x12.\n\x08unknown6\x18\x11 \x02(\x0b\x32\x1c.google.protobuf.StringValue\x12\'\n\x04geos\x18\x12 \x03(\x0b\x32\x19.LibraryDocument.Book.Geo\x12.\n\x08unknown7\x18\x13 \x02(\x0b\x32\x1c.google.protobuf.StringValue\x12,\n\x06rating\x18\x14 \x02(\x0b\x32\x1c.LibraryDocument.Book.Rating\x12/\n\x06genres\x18\x15 \x03(\x0b\x32\x1f.LibraryDocument.Book.BookGenre\x1a\x37\n\tBookGenre\x12*\n\x05genre\x18\x01 \x02(\x0b\x32\x1b.LibraryDocument.Book.Genre\x1a\"\n\x03Geo\x12\x0b\n\x03geo\x18\x01 \x02(\t\x12\x0e\n\x06geo_id\x18\x02 \x02(\t\x1a(\n\x05Genre\x12\r\n\x05genre\x18\x01 \x02(\t\x12\x10\n\x08subgenre\x18\x02 \x01(\t\x1a\x33\n\x06Rating\x12\x0e\n\x06rating\x18\x01 \x02(\x02\x12\x19\n\x11number_of_reviews\x18\x02 \x02(\t\x1am\n\x06Series\x12\x17\n\x0fthis_book_title\x18\x01 \x02(\t\x12\x32\n\x06series\x18\x02 \x03(\x0b\x32\".LibraryDocument.Book.SingleSeries\x12\x16\n\x0e\x62ook_in_series\x18\x03 \x02(\t\x1a\x9c\x01\n\x0cSingleSeries\x12\x11\n\tseries_id\x18\x01 \x02(\t\x12\x10\n\x08unknown1\x18\x02 \x02(\x05\x12\x10\n\x08unknown2\x18\x03 \x02(\x05\x12.\n\x08unknown3\x18\x04 \x02(\x0b\x32\x1c.google.protobuf.StringValue\x12\x10\n\x08unknown4\x18\x05 \x02(\x05\x12\x13\n\x0bseries_name\x18\x06 \x02(\t\x1aV\n\nCoverImage\x12\x0b\n\x03url\x18\x01 \x02(\t\x12\r\n\x05scale\x18\x02 \x02(\x05\x12\r\n\x05\x63olor\x18\x03 \x02(\t\x12\r\n\x05width\x18\x04 \x02(\x05\x12\x0e\n\x06height\x18\x05 \x02(\x05')

_globals = globals()
_builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, _globals)
_builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'google_internal_apis.input_pb2', _globals)
if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  _globals['_TAG']._serialized_start=101
  _globals['_TAG']._serialized_end=184
  _globals['_TAGGED']._serialized_start=186
  _globals['_TAGGED']._serialized_end=274
  _globals['_TAGSRESPONSE']._serialized_start=276
  _globals['_TAGSRESPONSE']._serialized_end=341
  _globals['_TAGREQUEST']._serialized_start=343
  _globals['_TAGREQUEST']._serialized_end=386
  _globals['_LIBRARYDOCUMENTRESPONSE']._serialized_start=388
  _globals['_LIBRARYDOCUMENTRESPONSE']._serialized_end=458
  _globals['_LIBRARYDOCUMENT']._serialized_start=461
  _globals['_LIBRARYDOCUMENT']._serialized_end=1677
  _globals['_LIBRARYDOCUMENT_BOOK']._serialized_start=535
  _globals['_LIBRARYDOCUMENT_BOOK']._serialized_end=1677
  _globals['_LIBRARYDOCUMENT_BOOK_BOOKGENRE']._serialized_start=1133
  _globals['_LIBRARYDOCUMENT_BOOK_BOOKGENRE']._serialized_end=1188
  _globals['_LIBRARYDOCUMENT_BOOK_GEO']._serialized_start=1190
  _globals['_LIBRARYDOCUMENT_BOOK_GEO']._serialized_end=1224
  _globals['_LIBRARYDOCUMENT_BOOK_GENRE']._serialized_start=1226
  _globals['_LIBRARYDOCUMENT_BOOK_GENRE']._serialized_end=1266
  _globals['_LIBRARYDOCUMENT_BOOK_RATING']._serialized_start=1268
  _globals['_LIBRARYDOCUMENT_BOOK_RATING']._serialized_end=1319
  _globals['_LIBRARYDOCUMENT_BOOK_SERIES']._serialized_start=1321
  _globals['_LIBRARYDOCUMENT_BOOK_SERIES']._serialized_end=1430
  _globals['_LIBRARYDOCUMENT_BOOK_SINGLESERIES']._serialized_start=1433
  _globals['_LIBRARYDOCUMENT_BOOK_SINGLESERIES']._serialized_end=1589
  _globals['_LIBRARYDOCUMENT_BOOK_COVERIMAGE']._serialized_start=1591
  _globals['_LIBRARYDOCUMENT_BOOK_COVERIMAGE']._serialized_end=1677
# @@protoc_insertion_point(module_scope)
