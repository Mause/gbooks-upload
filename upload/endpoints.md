"https://waa-pa.clients6.google.com/$rpc/google.internal.waa.v1.Waa/Ping"

"https://people-pa.clients6.google.com/$rpc/google.internal.people.v2.InternalPeopleService/GetPeople" [["me"],[[["person.name","person.email","person.photo"]],null,[5,1,7]]]

"/google.internal.play.books.audiobook.v1.AudiobookService/GetAudiobookResource"
"/google.internal.play.books.audiobook.v1.AudiobookService/GetAudiobookSupplement"

"/google.internal.play.books.cloudloading.v1.CloudLoadingOnePlatformService/Delete"
"/google.internal.play.books.cloudloading.v1.CloudLoadingOnePlatformService/Insert"

"/google.internal.play.books.enterprise.v1.EnterpriseService/AddClassroomBooksAttachment"
"/google.internal.play.books.enterprise.v1.EnterpriseService/AddLicenseOwners"
"/google.internal.play.books.enterprise.v1.EnterpriseService/AddLicenseRedeemers"
"/google.internal.play.books.enterprise.v1.EnterpriseService/ClaimLicense"
"/google.internal.play.books.enterprise.v1.EnterpriseService/GetLicenseHistory"
"/google.internal.play.books.enterprise.v1.EnterpriseService/GetLicenses" [null,null,null,1]
"/google.internal.play.books.enterprise.v1.EnterpriseService/GrantFreeBookLicenses"
"/google.internal.play.books.enterprise.v1.EnterpriseService/ListFreeBooksForLicense"
"/google.internal.play.books.enterprise.v1.EnterpriseService/RemoveLicenseOwners"
"/google.internal.play.books.enterprise.v1.EnterpriseService/RemoveLicenseRedeemers"

"/google.internal.play.books.family.v1.FamilyOnePlatformService/Share"
"/google.internal.play.books.family.v1.FamilyOnePlatformService/Unshare"

"/google.internal.play.books.layers.v1.VolumeAnnotationService/GetDictionaryDefinition" [null," unwrapping",[null,null,"en"]]

"/google.internal.play.books.library.v1.LibraryService/AddLibraryDocument"
"/google.internal.play.books.library.v1.LibraryService/AddTags"
"/google.internal.play.books.library.v1.LibraryService/CreateCustomTag"
"/google.internal.play.books.library.v1.LibraryService/DeleteCustomTag"
"/google.internal.play.books.library.v1.LibraryService/DeleteLibraryDocument"
"/google.internal.play.books.library.v1.LibraryService/ExpungeLibraryDocuments"
"/google.internal.play.books.library.v1.LibraryService/GetLibraryDocument"
"/google.internal.play.books.library.v1.LibraryService/GetVolumeOverviews"
"/google.internal.play.books.library.v1.LibraryService/HideLibraryDocuments"
"/google.internal.play.books.library.v1.LibraryService/ListTags"
"/google.internal.play.books.library.v1.LibraryService/LookupCopyLimit"
"/google.internal.play.books.library.v1.LibraryService/RemoveTags"
"/google.internal.play.books.library.v1.LibraryService/SyncDocumentPosition" ["84XKNwAAAEAJ",["384DC38E-1A56-4F48-8AE1-1C3F446F41E5",null,"1737093647557",[null,null,[2]],null,[["GBS.PA68.w.0.0.0.2"],null,0.9066666666666666]]]
"/google.internal.play.books.library.v1.LibraryService/SyncExtendedLibrary"
"/google.internal.play.books.library.v1.LibraryService/SyncUserLibrary"
"/google.internal.play.books.library.v1.LibraryService/UnhideLibraryDocuments"
"/google.internal.play.books.library.v1.LibraryService/UpdateCopyLimit"
"/google.internal.play.books.library.v1.LibraryService/UpdateCustomTag"

"/google.internal.play.books.series.v1.SeriesOnePlatformService/Fetch" [[["u4P3GgAAABBNmM"],["PL4pGwAAABAUpM"],["bW8uGwAAABBCdM"],["i1YuGwAAABCkTM"],["crbMGgAAABC_rM"]]]
"/google.internal.play.books.series.v1.SeriesOnePlatformService/FetchMembers"

"/google.internal.play.books.settings.v1.SettingsOnePlatformService/GetUserSettings"
"/google.internal.play.books.settings.v1.SettingsOnePlatformService/UpdateUserSettings"

"/google.internal.play.books.userannotations.v1.UserAnnotationService/CreateAnnotations"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/CreateCollection"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/DeleteAnnotations"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/DeleteCollection"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/ListAnnotations" [[["84XKNwAAAEAJ",null,"full-1.0.0"]],[1,2],null,1,1]
"/google.internal.play.books.userannotations.v1.UserAnnotationService/ListSharedAnnotations" [["84XKNwAAAEAJ",null,"full-1.0.0"],1,null,null,1]
"/google.internal.play.books.userannotations.v1.UserAnnotationService/TriggerExport"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/UpdateAnnotations"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/UpdateCollection"
"/google.internal.play.books.userannotations.v1.UserAnnotationService/UpdateSubscriptions"

"https://playgateway-pa.clients6.google.com/$rpc/play.gateway.adapter.books.v1.PlayGatewayBooksService/GetInAppStream" [[4,null,["84XKNwAAAEAJ",1]],null,null,null,null,[null,null,null,null,null,null,[3]],null,null,[null,[2],[1]]]
