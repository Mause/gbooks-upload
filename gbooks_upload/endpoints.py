from datetime import datetime
from pprint import pformat

from google.protobuf.timestamp_pb2 import Timestamp
from google.protobuf.wrappers_pb2 import StringValue

from google_internal_apis import LibraryServiceRpc
from input_pb2 import TagsResponse


def repeated(message, field, value):
    if field.type == field.TYPE_MESSAGE:
        for array in value:
            parse(array, getattr(message, field.name).add())
    else:
        getattr(message, field.name).extend(value)


def required(message, field, value):
    if field.type == field.TYPE_MESSAGE:
        parse(value, getattr(message, field.name))
    else:
        setattr(message, field.name, value)


def optional(message, field, value):
    if field.type == field.TYPE_MESSAGE:
        parse(value, getattr(message, field.name))
    else:
        setattr(message, field.name, value)


def parse(arrays, message):
    fields = message.DESCRIPTOR.fields
    if isinstance(message, Timestamp):
        message.FromMilliseconds(int(arrays))
        return message
    if isinstance(message, StringValue):
        if arrays:
            message.MergeFromString(arrays)
        return message

    for field in fields:
        value = arrays[field.number - 1]

        match field.label:
            case field.LABEL_REPEATED:
                repeated(message, field, value)
            case field.LABEL_REQUIRED:
                required(message, field, value)
            case field.LABEL_OPTIONAL:
                optional(message, field, value)
            case _:
                raise ValueError("Unknown label")
    remaining = arrays[fields[-1].number :]
    if remaining:
        raise ValueError("Extra fields: " + pformat(remaining))
    return message


class LibraryService(LibraryServiceRpc):
    async def add_tags(self, book_ids, tag_id):
        return await super().add_tags(
            [
                [
                    [book_id, tag_id, str(int(datetime.now().timestamp() * 1000))]
                    for book_id in book_ids
                ]
            ]
        )

    async def list_tags(self):
        res = parse(await super().list_tags(), TagsResponse())

        return {
            "tags": {tag.name: tag.tag_id for tag in res.tags},
            "tagged": [
                {
                    "book_id": tagged.book_id,
                    "tag_id": tagged.tag_id,
                    "tagged_at": tagged.tagged_at.ToDatetime(),
                }
                for tagged in res.tagged_items
            ],
        }

    async def get_library_document(self, book_id):
        return await super().get_library_document([[], book_id])
