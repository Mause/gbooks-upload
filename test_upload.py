import os

from click.testing import CliRunner

from gbooks_upload import main as upload
from gbooks_upload.endpoints import parse
from input_pb2 import TagsResponse

os.environ["TERM"] = "dumb"


def test_hello_world(snapshot):
    runner = CliRunner()
    result = runner.invoke(upload, ["--help"])
    assert result.exit_code == 0
    assert result.output == snapshot


def test_protoc(snapshot):
    assert parse([[["tag_name", "tag_id", 1]], []], TagsResponse()) == snapshot
