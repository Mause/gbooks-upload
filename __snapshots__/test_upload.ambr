# serializer version: 1
# name: test_hello_world
  '''
                                                                                  
   Usage: main [OPTIONS] COMMAND [ARGS]...                                        
                                                                                  
  ╭─ Options ────────────────────────────────────────────────────────────────────╮
  │ --help      Show this message and exit.                                      │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  ╭─ Commands ───────────────────────────────────────────────────────────────────╮
  │ books                                                                        │
  │ rpc                                                                          │
  │ shelves                                                                      │
  │ steal        Steal the cookie from a Chrome net-export log                   │
  │ upload       Upload files to Google Books                                    │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  
  '''
# ---
# name: test_protoc
  tags {
    name: "tag_name"
    tag_id: "tag_id"
    created_at {
      seconds: 1
    }
  }

# ---
